bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/DateTimeType.java

```diff
@@ -20,53 +22,83 @@
+     */
+    @Deprecated
+    public DateTimeType(Calendar calendar) {
+        this.zonedDateTime = ZonedDateTime.ofInstant(calendar.toInstant(), TimeZone.getDefault().toZoneId())
+                .withFixedOffsetZone();
+    }
 
     public DateTimeType() {
-        this(Calendar.getInstance());
+        this(ZonedDateTime.now());
     }
 
-    public DateTimeType(Calendar calendar) {
-        this.calendar = (Calendar) calendar.clone();
+    public DateTimeType(ZonedDateTime zoned) {
+        this.zonedDateTime = ZonedDateTime.from(zoned).withFixedOffsetZone();
     }
 
-    public DateTimeType(String calendarValue) {
-        Date date = null;
+    public DateTimeType(String zonedValue) {
+        ZonedDateTime date = null;
 
         try {
             try {
-                date = new SimpleDateFormat(DATE_PATTERN_WITH_TZ_AND_MS).parse(calendarValue);
-            } catch (ParseException fpe3) {
+                date = ZonedDateTime.parse(zonedValue, formatterTzMsRFC);
+            } catch (DateTimeParseException tzMsRfcException) {
                 try {
-                    date = new SimpleDateFormat(DATE_PATTERN_WITH_TZ_AND_MS_ISO).parse(calendarValue);
-                } catch (ParseException fpe4) {
+                    date = ZonedDateTime.parse(zonedValue, formatterTzMs);
+                } catch (DateTimeParseException tzMsException) {
                     try {
-                        date = new SimpleDateFormat(DATE_PATTERN_WITH_TZ).parse(calendarValue);
-                    } catch (ParseException fpe2) {
-                        date = new SimpleDateFormat(DATE_PATTERN).parse(calendarValue);
+                        date = ZonedDateTime.parse(zonedValue, formatterTz);
+                    } catch (DateTimeParseException tzException) {
+                        try {
+                            date = ZonedDateTime.parse(zonedValue, formatter);
+                        } catch (DateTimeParseException regularFormatException) {
+                            date = ZonedDateTime.parse(zonedValue, formatterTzMsIso);
+                        }
                     }
                 }
             }
-        } catch (ParseException fpe) {
-            throw new IllegalArgumentException(calendarValue + " is not in a valid format.", fpe);
+        } catch (DateTimeParseException invalidFormatException) {
+            throw new IllegalArgumentException(zonedValue + " is not in a valid format.", invalidFormatException);
         }
 
         if (date != null) {
-            calendar = Calendar.getInstance();
-            calendar.setTime(date);
+            zonedDateTime = ZonedDateTime.from(date);
         }
     }
 
+    @Deprecated
     public Calendar getCalendar() {
-        return (Calendar) calendar.clone();
+        return GregorianCalendar.from(zonedDateTime);
+    }
+
+    public ZonedDateTime getZonedDateTime() {
+        return ZonedDateTime.of(zonedDateTime.toLocalDateTime(), zonedDateTime.getZone());

```

Comment:
Why not simple `return zonedDateTime;`? We are using an immutable object here and there is no need to create duplicated of it.